homeassistant/components/bzutech/config_flow.py

```diff
@@ -0,0 +1,264 @@
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    hass.data[DOMAIN] = {}
+    api = BzuTech(data[CONF_EMAIL], data[CONF_PASSWORD])
+
+    hass.data[DOMAIN]["bzuapi"] = api
+
+    if not await api.start():
+        raise InvalidAuth
+
+    return {"title": "BZUTech", "bzuapi": api}
+
+
+def get_devices(hass: HomeAssistant, page: int) -> dict[str, Any]:
+    """Get device names on a dict for the showmenu."""
+
+    devices = {}
+    i = 1
+    first = page * 4
+    last = first + 3
+    counter = 0
+
+    for key in list(hass.data[DOMAIN]["bzuapi"].dispositivos.keys()):
+        if first <= counter <= last:
+            returnkey = "option" + str(i)
+            devices[returnkey] = key
+            i = i + 1
+
+        counter = counter + 1
+    if len(list(hass.data[DOMAIN]["bzuapi"].dispositivos.keys())) > (page + 1) * 4:
+        devices["option5"] = "Mais dispositivos"
+    return devices
+
+
+def get_sensors(
+    hass: HomeAssistant, devicepos: int, sensorport: int, page: int
+) -> dict[str, Any]:
+    """Get sensor names from a device on a dict for the showmenu."""
+
+    sensors = {}
+    first = page * 4
+    last = first + 3
+    counter = 0
+    chipid = list(hass.data[DOMAIN]["bzuapi"].dispositivos.keys())[devicepos]
+    numSensors = len(
+        list(
+            hass.data[DOMAIN]["bzuapi"]
+            .dispositivos[chipid]
+            .get_sensor_names_on(str(sensorport))
+        )
+    )
+
+    i = 1
+
+    for name in list(
+        hass.data[DOMAIN]["bzuapi"]
+        .dispositivos[chipid]
+        .get_sensor_names_on(str(sensorport))
+    ):
+        if first <= counter <= last:
+            sensors["option" + str(i)] = name
+            i = i + 1
+        counter = counter + 1
+
+    if numSensors > 4 * (page + 1):
+        sensors["option5"] = "Outros Sensores"
+
+    return sensors
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for BZUTech."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await validate_input(self.hass, user_input)
+                self.hass.data[DOMAIN][CONF_EMAIL] = user_input[CONF_EMAIL]
+                self.hass.data[DOMAIN][CONF_PASSWORD] = user_input[CONF_PASSWORD]
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                self.hass.data[DOMAIN]["disppage"] = 0
+                self.hass.data[DOMAIN]["sensorpage"] = 0
```

Comment:
just use a class attribute to store those instead of writing them to `hass.data`